File: web/src/services/cloud-account.ts
import request from '../utils/request'
import type { PageResult } from './audit'

export interface CloudAccount {
  id: number
  name: string
  provider: string
  access_key_id: string
  access_key_secret: string
  region: string
  status: number
  last_sync_at: string
  last_sync_status: string
  created_at: string
}

export interface CreateCloudAccountParams {
  name: string
  provider: string
  access_key_id: string
  access_key_secret: string
  region?: string
}

export interface UpdateCloudAccountParams {
  name?: string
  access_key_id?: string
  access_key_secret?: string
  region?: string
}

export function getCloudAccounts(params?: Record<string, unknown>): Promise<PageResult<CloudAccount>> {
  return request.get('/cloud-accounts', { params })
}

export function getCloudAccount(id: number): Promise<CloudAccount> {
  return request.get(`/cloud-accounts/${id}`)
}

export function createCloudAccount(data: CreateCloudAccountParams): Promise<void> {
  return request.post('/cloud-accounts', data)
}

export function updateCloudAccount(id: number, data: UpdateCloudAccountParams): Promise<void> {
  return request.put(`/cloud-accounts/${id}`, data)
}

export function deleteCloudAccount(id: number): Promise<void> {
  return request.delete(`/cloud-accounts/${id}`)
}

File: internal/model/cloud_account.go
package model

import "time"

// CloudAccount 云账号模型
type CloudAccount struct {
	BaseModel
	Name            string     `gorm:"size:128;not null" json:"name"`
	Provider        string     `gorm:"size:32;not null;default:aliyun" json:"provider"` // aliyun
	AccessKeyID     string     `gorm:"size:128;not null" json:"access_key_id"`
	AccessKeySecret string     `gorm:"size:512;not null" json:"-"` // AES-GCM 加密存储
	Region          string     `gorm:"size:64" json:"region"`
	Status          int        `gorm:"default:1" json:"status"` // 1=启用 0=禁用
	LastSyncAt      *time.Time `json:"last_sync_at"`
	LastSyncStatus  string     `gorm:"size:32" json:"last_sync_status"` // success / failed / syncing
	CreatedBy       uint       `json:"created_by"`
}
